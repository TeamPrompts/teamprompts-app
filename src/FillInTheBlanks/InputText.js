import React, { useState } from 'react';

function InputText({ placeholder }) {
  const [size, setSize] = useState(placeholder.length);

  function onChange(value) {
    if (value.length > placeholder.length) {
      setSize(value.length);
    } else {
      setSize(placeholder.length);
    }
  }

  return (
    <input
      data-testid="input"
      onChange={e => onChange(e.target.value)}
      placeholder={placeholder}
      size={size}
      type="text"
    />
  );
}

export default InputText;
