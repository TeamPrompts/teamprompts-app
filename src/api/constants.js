export const ACTIVE = 'active';
export const ASC = 'asc';
export const AUTHOR = 'Author';
export const DESC = 'desc';
export const EXAMPLES = 'Examples';
export const FITB = 'FITB';
export const FITBS = 'FITBs';
export const ID = 'ID';
export const NAME = 'Name';
export const PROMPTS = 'Prompts';
export const STATUS = 'Status';
export const TAGS = 'Tags';
