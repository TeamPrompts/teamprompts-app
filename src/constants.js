// XS (4), S (8), M (16), L (32)

export const BLANK = '________________'; // M
export const BLANK_XS = '____';
export const MASTER_DETAIL_V1 = 'MASTER_DETAIL_V1';
export const NONE = 'NONE';
export const SEPARATOR = '__BLANK__';
export const TEAMPROMPTS_APP_EXPERIMENT = 'TEAMPROMPTS_APP_EXPERIMENT';

export const modes = {
  blanks: 'blanks',
  blanksXS: 'blanksXS',
  examples: 'examples',
  input: 'input',
  prompts: 'prompts'
};

export const pageTypes = {
  CategoryPage: 'CategoryPage',
  HomePage: 'HomePage',
  PromptsPage: 'PromptsPage'
};
