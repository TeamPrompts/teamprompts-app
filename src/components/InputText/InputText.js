import classname from 'classnames';
import React from 'react';
import AutosizeInput from 'react-input-autosize';
import { BLANK } from '../../constants';
import { CHANGE } from '../../FillInTheBlanks/reducer';

function InputText({ dispatch, hint, id, value }) {
  function onChange({ target: { value } }) {
    dispatch({ type: CHANGE, value: { id, value: value } });
  }

  return (
    <span className="inline-flex flex-col leading-none">
      <AutosizeInput
        className="self-center"
        data-testid="input"
        inputClassName={classname({ 'bg-purple-100': value.length > 0 })}
        onChange={onChange}
        placeholder={BLANK}
        placeholderIsMinWidth={true}
        value={value}
      />
      {hint !== BLANK && <label className="self-center text-sm">{hint}</label>}
    </span>
  );
}

export default InputText;
