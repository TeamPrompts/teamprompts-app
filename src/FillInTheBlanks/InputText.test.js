import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import { BLANK } from '../constants';
import InputText from './InputText';

describe('InputText', () => {
  const props = {
    dispatch: jest.fn(),
    hint: BLANK,
    id: 0,
    value: ''
  };

  it('to match snapshot', () => {
    const tree = create(<InputText {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('on change', () => {
    const { getByTestId } = render(<InputText {...props} />);
    const ID = 'input';
    const LOREM_IPSUM = 'lorem ipsum dolor sit amet';

    fireEvent.change(getByTestId(ID), { target: { value: LOREM_IPSUM } });
    expect(props.dispatch).toBeCalled();
  });
});
