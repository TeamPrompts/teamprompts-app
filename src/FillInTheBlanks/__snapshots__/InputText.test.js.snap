// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText to match snapshot 1`] = `
<input
  data-testid="input"
  onChange={[Function]}
  placeholder="________________"
  size={16}
  type="text"
/>
`;
